packages/sync/src/links/LocalDirectiveFilterLink.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ApolloLink,
+  NextLink,
+  Observable,
+  Operation
+} from "apollo-link";
+import { hasDirectives, removeDirectivesFromDocument, checkDocument } from "apollo-utilities";
+import { LocalDirectivesArray, MUTATION_QUEUE_LOGGER, LocalDirectives } from "../config/Constants";
+import debug from "debug";
+
+export const logger = debug(MUTATION_QUEUE_LOGGER);
+
+export class LocalDirectiveFilterLink extends ApolloLink {
+  private connectionRemoveOnlineOnly = {
```

Comment:
If we do not use array to build those then this array makes no sense as there is still manual input needed.